examples/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,124 @@
+module AffinePlaneCurveModule
+
+using Oscar, Markdown
+
+export AffinePlaneCurve, AffinePlaneCurve_degree, AffinePlaneCurve_Jacobian, AffinePlaneCurve_tangent, AffinePlaneCurve_issmoothpoint, AffinePlaneCurve_union
+
+####################################################
+#
+#We follow the lecture notes by Andreas Gathmann: https://www.mathematik.uni-kl.de/~gathmann/en/curves.php
+#
+####################################################
+mutable struct AffinePlaneCurve{S}
+  eq::Oscar.MPolyElem                      #Equation of the curve (polynomial in two variables)
+  degree::Union{String, Int64}             #degree of the equation of the curve
```

Comment:
A comment here about what strings are allowed would be useful.

You might consider defining a special type for the degree rather than using a Union, as it's going to get cumbersome to add "if" statements everywhere to handle both cases.